web-registry/src/features/marketplace/BridgeFlow/BridgeFlow.tsx

```diff
@@ -0,0 +1,155 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { DeliverTxResponse } from '@cosmjs/stargate';
+import { useTrack } from 'use-analytics';
+
+import ErrorBanner from 'web-components/lib/components/banner/ErrorBanner';
+import { CelebrateIcon } from 'web-components/lib/components/icons/CelebrateIcon';
+import ErrorIcon from 'web-components/lib/components/icons/ErrorIcon';
+import { ProcessingModal } from 'web-components/lib/components/modal/ProcessingModal';
+import { TxErrorModal } from 'web-components/lib/components/modal/TxErrorModal';
+import { Item } from 'web-components/lib/components/modal/TxModal';
+import { TxSuccessfulModal } from 'web-components/lib/components/modal/TxSuccessfulModal';
+
+import { BatchInfoWithBalance } from 'types/ledger/ecocredit';
+import { UseStateSetter } from 'types/react/use-state';
+import { getHashUrl } from 'lib/block-explorer';
+
+import { Link } from 'components/atoms';
+import { BridgeModal } from 'components/organisms';
+import { useMsgClient } from 'hooks';
+
+import { BRIDGE_HEADER } from './BridgeFlow.constants';
+import useCreditBridgeSubmit from './hooks/useCreditBridgeSubmit';
+
+type Props = {
+  isFlowStarted: boolean;
+  setBatchToBridge: UseStateSetter<BatchInfoWithBalance | undefined>;
+  selectedBatch: BatchInfoWithBalance | undefined;
+};
+
+export const BridgeFlow = ({
```

Comment:
Our flows have a lot of code in common so far. Starting to think that we could reduce the boilerplate by extracting modals into the layout.
Nothing to change here but since we'll probably write more flow like this in the future it might be worth investigating.